### ISSUE TITLE
Loss of performance

### ISSUE THREAD

**Issue by karamavusibrahim** [webdataset/webdataset#306]
I am trying to use webdataset instead of my custom dataset.However,i am getting loss of performance at training process and also memory usage of GPUs were increased .Is it normal? These are my environment and my benchmark results(I used 2 A100):

CUDA 11.8
      - accelerate==0.24.0
      - braceexpand==0.1.7
      - certifi==2022.12.7
      - charset-normalizer==2.1.1
      - diffusers==0.21.4
      - filelock==3.9.0
      - fsspec==2023.4.0
      - huggingface-hub==0.17.3
      - idna==3.4
      - imageio==2.31.6
      - importlib-metadata==6.8.0
      - jinja2==3.1.2
      - lazy-loader==0.3
      - markupsafe==2.1.2
      - mpmath==1.3.0
      - natsort==8.4.0
      - networkx==3.0
      - numpy==1.24.1
      - opencv-python==4.8.1.78
      - packaging==23.2
      - pandas==2.1.1
      - pillow==9.3.0
      - psutil==5.9.6
      - python-dateutil==2.8.2
      - pytz==2023.3.post1
      - pyyaml==6.0.1
      - regex==2023.10.3
      - requests==2.28.1
      - safetensors==0.4.0
      - scikit-image==0.22.0
      - scipy==1.11.3
      - six==1.16.0
      - sympy==1.12
      - tifffile==2023.9.26
      - tokenizers==0.14.1
      - torch==2.1.0+cu118
      - torchaudio==2.1.0+cu118
      - torchvision==0.16.0+cu118
      - tqdm==4.66.1
      - transformers==4.34.1
      - triton==2.1.0
      - typing-extensions==4.4.0
      - tzdata==2023.3
      - urllib3==1.26.13
      - webdataset==0.2.62
      - zipp==3.17.0

Custom usage:
Iteration/second for one step:6.94s/it
Memory usages: 78234 MiB 78235 MiB

Webdataset(Custom Dataset with IterableDataset):
Iteration/second for one step:7.62s/it
Memory usages: 79181 MiB 79091 MiB

Webdataset used as below:
wds.DataPipeline(
        wds.ResampledShards(url),
        wds.tarfile_to_samples(),
        wds.shuffle(1000),
        wds.decode("pil",handler=wds.warn_and_continue),
        wds.to_tuple("jpg;png"),
        wds.map(preprocess_wds),
        #wds.batched(args.train_batch_size,partial=False,collation_fn=collate_fn),
        
    ).with_epoch(10000)
Iteration/second for one step:7.42s/it
Memory usages: 79049 MiB 79209 MiB

Webdataset used as below(and also tar files are in bucket):
wds.DataPipeline(
        wds.ResampledShards(url),
        wds.tarfile_to_samples(),
        wds.shuffle(1000),
        wds.decode("pil",handler=wds.warn_and_continue),
        wds.to_tuple("jpg;png"),
        wds.map(preprocess_wds),
        #wds.batched(args.train_batch_size,partial=False,collation_fn=collate_fn),
        
    ).with_epoch(10000)
Iteration/second for one step:7.66s/it
Memory usages: 79052 MiB 79220 MiB

Webdataset used as below:
wds.WebDataset(url,resampled=True).shuffle(1000)
    train_dataset = (train_dataset.decode("pil",handler=wds.warn_and_continue).to_tuple("jpg;png"))
    train_dataset = train_dataset.map(preprocess_wds)
    #train_dataset = train_dataset.batched(batchsize=args.train_batch_size,partial=False,
    #                                           collation_fn=collate_fn)
    train_dataset = train_dataset.with_epoch(10000)
Iteration/second for one step:7.65s/it
Memory usages: 79237 MiB 79237 MiB

I can not use WebLoader because accelerate library is not supporting WebLoader.Dataloader config is same for all usages.Can you help?
